beacon_chain/state/state_transition.py

```diff
@@ -17,32 +40,28 @@
     get_signed_parent_hashes,
 )
 
-import beacon_chain.utils.bls as bls
-from beacon_chain.utils.blake import (
-    blake,
-)
-from beacon_chain.utils.bitfield import (
-    get_bitfield_length,
-    has_voted,
-)
-from beacon_chain.utils.simpleserialize import (
-    deepcopy,
-)
+if TYPE_CHECKING:
+    from .attesation_record import AttestationRecord
+    from .block import Block
 
+BlockVoteCache = Dict[str, Dict[str, Union[str, bytes, int]]]
+Hash32 = NewType('Hash32', bytes)
+ShardId = NewType('ShardId', int)
 
-def validate_block(block):
+
+def validate_block(block: 'Block') -> bool:
     # ensure parent processed
     # ensure pow_chain_ref processed
     # ensure local time is large enough to process this block's slot
 
     return True
 
 
-def validate_attestation(crystallized_state,
-                         active_state,
-                         attestation,
-                         block,
-                         config=DEFAULT_CONFIG):
+def validate_attestation(crystallized_state: CrystallizedState,
+                         active_state: ActiveState,
+                         attestation: 'AttestationRecord',
+                         block: 'Block',
```

Comment:
No quotes necessary

Reply:
It's because I'm using `TYPE_CHECKING`: https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING
Although we haven't separated the test env and product installations.